src/VisibleComponent/VisibleComponent.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Logger from '../Util/Logger';
+
+/**
+ * The HOC factory function.
+ *
+ * Wrapped components will be checked against the activeModules array of
+ * the state: If the wrapped component (identified by it's name) is included
+ * in the state, it will be rendered, if not, it wont.
+ *
+ * @param {Component} WrappedComponent The component to wrap and enhance.
+ * @return {Component} The wrapped component.
```

Comment:
I think you missed another `@param` here, right?

Reply:
Check.